app/models/king.rb

```diff
@@ -12,6 +12,39 @@ def valid_move?(x, y)
     return false if blocked?(x, y) # checks if piece is blocking at destination coordinate
     return false if delta_x(x_position, x) > 1
     return false if delta_y(y_position, y) > 1
+
+    # The king can castle
+    return true if move_castle?(x, y)
+
     true
   end
+
+  # Support castling with the king
+  def move_castle?(x, y)
+    castling_kingside?(x, y) && castling_never_moved?
+  end
+
+  # The king can attack to the castle on the king side
+  def castling_kingside?(x, y)
+    x == 6 && y == y_position && castling_rook_kingside && castling_rook_kingside.castling_never_moved? && \
+      castling_rook_kingside.blocked?(4, y_position) == false
+  end
+
+  # The rook to castle with king side
+  def castling_rook_kingside
+    can_castling_rook_kingside = game.pieces.where(x_position: 0, y_position: y_position, \
+                                                   color: 'white' || 'black', type: 'Rook')
+    @castling_rook_kingside = can_castling_rook_kingside.first
```

Comment:
I don't think this needs to be saved to an instance variable, so this line could instead just be:

```ruby
can_castling_rook_kingside.first
```

However then the entire method could just be the query itself:

```ruby
def castling_rook_kingside
  game.pieces.where(x_position: 0, y_position: y_position, color: color, type: 'Rook').first
end
```